src/Compilers/VisualBasic/Test/Semantic/Compilation/SemanticModelGetDeclaredSymbolAPITests.vb

```diff
@@ -2128,6 +2128,59 @@ End Module
                 Diagnostic(ERRID.ERR_IllegalChar, "$"))
         End Sub
 
+        <WorkItem(59709, "https://github.com/dotnet/roslyn/issues/59709")>
+        <Fact>
+        Public Sub LocalUsing()
+            Dim compilation = CompilationUtils.CreateCompilationWithMscorlib40(
+<compilation>
+    <file name="a.vb">
+Module M1
+    Class C
+        Sub S() As Object
+            Using writer As System.IO.TextWriter = System.IO.File.CreateText("log.txt")
+                writer.WriteLine("")    'BIND:"writer"
+            End Using
+        End Sub
+    End Class
+End Module
+    </file>
+</compilation>)
+
+            Dim model = GetSemanticModel(compilation, "a.vb")
+            Dim expressionSyntax = CompilationUtils.FindBindingText(Of IdentifierNameSyntax)(compilation, "a.vb", 0)
+            Dim local = DirectCast(model.GetSymbolInfo(expressionSyntax).Symbol, LocalSymbol)
+            Assert.False(local.IsConst)
+            Assert.False(local.IsForEach)
+            Assert.True(local.IsUsing)
+        End Sub
+
+        <WorkItem(59709, "https://github.com/dotnet/roslyn/issues/59709")>
+        <Fact>
+        Public Sub LocalForEach()
+            Dim compilation = CompilationUtils.CreateCompilationWithMscorlib40(
+<compilation>
+    <file name="a.vb">
+Module M1
+    Class C
+        Sub S() As Object
+            Dim a() As Integer = {1, 2, 3}
+            For Each x As Integer In a
+                Dim y = x   'BIND:"x"
+            Next
+        End Sub
+    End Class
+End Module
+    </file>
+</compilation>)
+
+            Dim model = GetSemanticModel(compilation, "a.vb")
+            Dim expressionSyntax = CompilationUtils.FindBindingText(Of IdentifierNameSyntax)(compilation, "a.vb", 0)
+            Dim local = DirectCast(model.GetSymbolInfo(expressionSyntax).Symbol, LocalSymbol)
+            Assert.False(local.IsConst)
+            Assert.True(local.IsForEach)
```

Comment:
>Assert.True(local.IsForEach) [](http://example.com/codeflow?start=12&length=28)

Consider adding an assert of this form ```Assert.True(local.IsForEach())```. Prior to addition of ILocalSymbol.IsForEach property, it would call an extension method from Microsoft.CodeAnalysis.VisualBasic.VisualBasicExtensions. After the property is added, it will call the property instead.

Reply:
71bbb2cc22a74c7384e10adef4f2393be42670aa